GPS/src/main.cpp

```diff
@@ -0,0 +1,194 @@
+#define SIZE_VALUES 255
+
+float getLatitude(char latData[], char latSign[]);
+int splitData(const char *pGpsData);
+float getLongitude(char longData[], char longSign[]);
+
+float latitude;
+float longitude;
+float fixType;
+
+void setup()
+{
+  Serial.begin(115200);
+  Serial1.begin(9600, SERIAL_8N1, RX1PIN, TX1PIN);
+
+  for (EVER)
+  {
+    char bufferGpsReceive[BUFFER_SIZE] = {0};
+    uint8_t dataSize = Serial1.readBytesUntil('\n', bufferGpsReceive, BUFFER_SIZE - 1);
+
+    if (dataSize > 0)
+    {
+      // Good Received data: $GNGGA,201427.00,4522.65273,N,07155.50339,W,2,12,0.71,282.0,M,-31.5,M,,0000*78
+      bufferGpsReceive[dataSize] = '\0';
+      const char *pGpsData = bufferGpsReceive;
+      char gpsDataTest[BUFFER_SIZE] = {'$', 'G', 'N', 'G', 'G', 'A', ',', '2', '0', '1', '4', '2', '7', '.', '0', '0', ',', '4', '5', '2', '2', '.', '6', '5', '2', '7', '3', ',', 'N', ',', '0', '7', '1', '5', '5', '.', '5', '0', '3', '3', '9', ',', 'W', ',', '2', ',', '1', '2', ',', '0', '.', '7', '1', ',', '2', '8', '2', '.', '0', ',', 'M', ',', '-', '3', '1', '.', '5', ',', 'M', ',', ',', '0', '0', '0', '0', '*', '7', '8', '\0'};
+      const char *pGpsDataTest = gpsDataTest;
+      splitData(pGpsData);
+    }
+  }
+}
+
+void loop() {}
+
+int splitData(const char *pGpsData)
+{
+  char data[NB_ELEMENTS][SIZE_VALUES] = {0};
+  uint8_t indexElements = 0;
+  uint8_t indexChar = 0;
+  bool valide = true;
+
+  if (pGpsData[0] == '\n' || pGpsData[0] == '\0' || pGpsData[0] != '$')
+  {
+    valide = false;
+    LOG(WARN, "Empty message or not formatted correctly");
+  }
+  else
+  {
+    uint16_t i = 0;
+    for (; pGpsData[i] != '\n' && pGpsData[i] != '\0' && i < UINT8_MAX; i++)
+    {
+      if (pGpsData[i] == ',')
+      {
+        data[indexElements][indexChar + 1] = '\0';
+        indexChar = 0;
+        indexElements++;
+      }
+      else
+      {
+        data[indexElements][indexChar] = pGpsData[i];
+        indexChar++;
+      }
+    }
+  }
+
+  char compareMsgType[] = "$GNGGA";
+  if (strcmp(data[0], compareMsgType) == 0)
+  {
+    LOG(INFO, "Good format of received data: %s", pGpsData);
+  }
+  else
+  {
+    valide = false;
+  }
+
+  fixType = atoi(data[6]);
+  if (fixType == 0)
+  {
+    valide = false;
+    LOG(WARN, "Invalid messages");
+  }
+
+  if (valide)
+  {
+    if (data[2][0] != '\0')
+    {
+      latitude = getLatitude(data[2], data[3]);
+      LOG(INFO, "Latitude: %f", latitude);
+    }
+    else
+    {
+      valide = false;
+      LOG(WARN, "Empty messages");
+    }
+
+    if (data[4][0] != '\0')
+    {
+      longitude = getLongitude(data[4], data[5]);
+      LOG(INFO, "Longitude: %f", longitude);
+    }
+    else
+    {
+      valide = false;
+      LOG(WARN, "Empty messages");
+    }
+  }
+
+  return 0;
+}
+
+float getLatitude(char latData[], char latSign[])
+{
+  float allDegrees = 0;
+  float allMinutes = 0;
+  uint8_t i = 0;
+  int8_t iDeg = 1;
+  int8_t iMin = 1;
+
+  for (; latData[i] != '\0'; i++)
```

Comment:
chances d'overshoot de l'array -> met une condition de sorti juste pour être sur de pas tombé dans une loop infinie, example:
for (; latData[i] != '\0' && i < MAX_SIZE;  i++)